uefi/firmwarevolume.go

```diff
@@ -189,6 +190,39 @@ func (fv *FirmwareVolume) Assemble() ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
+	// Make sure we don't go over the size.
+	fvLen := uint64(len(fv.buf))
+	fOffset := fv.DataOffset
+	for _, f := range fv.Files {
+		fBuf, err := f.Assemble()
+		if err != nil {
+			return nil, err
+		}
+		fLen := uint64(len(fBuf))
+		if fLen+fOffset > fvLen {
+			// TODO: Actually loop through and calculate the full size so we know how much to reduce by.
+			// For now we just return early
+			return nil, errors.New("insufficient space in FV, files too big")
+		}
+		// Overwrite old data in the firmware volume.
+		// We have to pad to the 8 byte alignments.
+		for alignBytes := align8(fOffset) - fOffset; alignBytes > 0; alignBytes-- {
+			fv.buf[fOffset] = 0xFF
+			fOffset++
+		}
+		copy(fv.buf[fOffset:], fBuf)
+		fOffset += fLen
+	}
+
+	if fOffset == fvLen {
```

Comment:
This if-statement doesn't look necessary because the following for-loop is skipped and the same values are returned when it is false.

Reply:
oo ok thanks!